### PR TITLE
enable DPLPMTUD on macOS dual-stack sockets

### PR DESCRIPTION
**Pull request by marten-seemann**
My new draft in the TSV WG (https://datatracker.ietf.org/doc/draft-seemann-tsvwg-udp-fragmentation/) contains details on how IP fragmentation is handled on different platforms.

### CODE REVIEW

**Review by sukunrt — APPROVED** [review]
This doesn't work for macOS Ventura(13.1). Here the DF bit won't be set and DPLPMTUD will set the max value of 1430. It's 1430 because this is 20 bytes less than the max MTU for ethernet frames. https://github.com/quic-go/quic-go/blob/macos-dplpmtud/mtu_discoverer.go#L25

**Comment by marten-seemann** [comment]
I updated the PR to test if a socket is a dual-stack socket on older (before Sequoia) macOS versions. If it's a dual-stack socket, DPLPMTUD will be disabled.

**Review by sukunrt — CHANGES_REQUESTED** [review]

File: sys_conn_df_darwin.go [hunk 1]
```diff
@@ -4,47 +4,61 @@ package quic
...
+		// See https://datatracker.ietf.org/doc/draft-seemann-tsvwg-udp-fragmentation/ for details.
+		switch addr.(type) {
+		case *unix.SockaddrInet4:
+			controlErr = unix.SetsockoptInt(int(fd), unix.IPPROTO_IP, unix.IP_DONTFRAG, 1)
+		case *unix.SockaddrInet6:
+			controlErr = unix.SetsockoptInt(int(fd), unix.IPPROTO_IPV6, unix.IPV6_DONTFRAG, 1)
+
+			// Setting the DF bit on dual-stack sockets works since macOS Sequoia.
+			// Disable DF on dual-stack sockets before Sequoia.
+			if version < 15 {
```
[the remark this answers is not in the document]
Comment (marten-seemann) [reply sys_conn_df_darwin.go]: Fixed by introducing a constant `macOSVersion15`.

File: sys_conn_df_darwin.go [hunk 2]
```diff
@@ -4,47 +4,61 @@ package quic
...
-	"github.com/quic-go/quic-go/internal/utils"
 )
 
 func setDF(rawConn syscall.RawConn) (bool, error) {
-	// Setting DF bit is only supported from macOS11
+	// Setting DF bit is only supported from macOS 11.
 	// https://github.com/chromium/chromium/blob/117.0.5881.2/net/socket/udp_socket_posix.cc#L555
-	if supportsDF, err := isAtLeastMacOS11(); !supportsDF || err != nil {
+	version, err := getMacOSVersion()
+	if err != nil || version < 11 {
```
[the remark this answers is not in the document]
Comment (marten-seemann) [reply sys_conn_df_darwin.go]: Fixed by introducing a constant `macOSVersion11`.

**Review by sukunrt — APPROVED** [review]

**Pull request merged**